### ISSUE TITLE
Upstart service provider doesn't work in Centos 6

### ISSUE THREAD

**Issue by intjonathan** [newrelic/puppet-heka#1]
A known [puppet bug](https://projects.puppetlabs.com/issues/14693) is triggered when trying to use the Service management, and results in this message:

```
Error: /Stage[main]/Heka::Service/Service[heka]: Provider upstart is not functional on this host
```

**Comment by nickchappell** [comment]
Which CentOS 6 point release did you see this in? Which Puppet version? I tested this on 6.6 and Puppet 3.7 (open source, not PE).

**Comment by intjonathan** [comment]
```
[vagrant@test-target-1 ~]$ puppet --version
3.3.1 (Puppet Enterprise 3.1.0)
[vagrant@test-target-1 ~]$ cat /etc/redhat-release
CentOS release 6.6 (Final)
```

**Comment by intjonathan** [comment]
Separately, if you try to use the workaround of specifying `class { '::heka': manage_service => false...`, you get:

```
Error: Could not find dependent Service[heka] for File[/etc/heka/heka.toml] at /tmp/vagrant-puppet/modules/heka/manifests/config.pp:45
```

I think due to the `notify => Service...` in [config.pp](https://github.com/newrelic/puppet-heka/blob/master/manifests/config.pp#L44) and many other places.

**Comment by nickchappell** [comment]
https://github.com/newrelic/puppet-heka/pull/8 makes management of the service optional, but I'll open up a separate PR for this once https://github.com/newrelic/puppet-heka/pull/8 is merged.

**Comment by nickchappell** [comment]
I opened https://github.com/newrelic/puppet-heka/pull/12 to fix this.

**Comment by nickchappell** [comment]
I closed #12 and created a new PR so it could merge cleanly: https://github.com/newrelic/puppet-heka/pull/15
